homework_2/task_2/test.c

```diff
@@ -0,0 +1,68 @@
+#define CTEST_MAIN
+#define CTEST_SEGFAULT
+#include "../../ctest/ctest.h"
+
+#include "stdio.h"
+#include "sortings.h"
+
+int main(int argc, const char *argv[]) {
+    return ctest_main(argc, argv);
+}
+
+void assertEqual(int *arrayA, int *arrayB, int size) {
+    for (int i = 0; i < size; ++i) {
+        ASSERT_EQUAL(arrayA[i], arrayB[i]);
+    }
+}
+
+typedef void (*sort)(int *, int);
+
+void testEmptyArray(sort sort) {
+    sort(NULL, 0);
+}
+
+void testOneElementArray(sort sort) {
+#define size 1
+    int array[size] = { 2354 },
+        expected[size] = { 2354 };
+    sort(array, size);
+    assertEqual(array, expected, size);
+#undef size
+}
+
+void testEightElementsArray(sort sort) {
+#define size 8
+    int array[size] = { 2865, -3452, 123, 0, 2104, -1923, -242, 1532 },
+        expected[size] = { -3452, -1923, -242, 0, 123, 1532, 2104, 2865 };
+    sort(array, size);
+    assertEqual(array, expected, size);
+#undef size
+}
+
+void test20ElementsSameValuesArray(sort sort) {
+#define size 20
+    int array[size], expected[size];
+
+    for (int i = 0; i < size; ++i) {
+        array[i] = expected[i] = 1532;
```

Comment:
По стайлгайду один оператор за раз, так что как-то так:
```suggestion
        array[i] = 1532;
        expected[i] = 1532;
```
Потому что так сходу не всякий скажет, что возвращает оператор присваивания в качестве значения (по логике вещей должен ничего).